Submitted By:     VMiklos <vmiklos@frugalware.org>
Date:             2005.03.05
Initial Packages: 2.9.5
Description:      This patch fixes an annoying bug in makepkg.
                  Instead of removing the documentation, add some automatically.

diff -Naur pacman-2.9.5.orig/scripts/makepkg pacman-2.9.5/scripts/makepkg
--- pacman-2.9.5.orig/scripts/makepkg	2004-12-19 03:09:59.000000000 +0100
+++ pacman-2.9.5/scripts/makepkg	2005-02-03 14:58:50.048936144 +0100
@@ -560,10 +560,36 @@
 	exit 2
 fi
 
-# remove info/doc files
+# documentation
+msg "Preparing package documentation..."
 cd $startdir
-rm -rf pkg/usr/info pkg/usr/share/info
-rm -rf pkg/usr/doc pkg/usr/share/doc
+if [ -d pkg/usr/doc ]; then
+	mkdir -p pkg/usr/share/doc
+	cp -a pkg/usr/doc/* pkg/usr/share/doc/
+	rm -rf pkg/usr/doc
+fi
+mkdir -p pkg/usr/share/doc/$pkgname-$pkgver
+for i in `find src -maxdepth 2 -name ANNOUNCE -o -name AUTHORS -o -name *BUGS* -o -name CHANGES -o -name CONFIGURATION -o -name *COPYING* -o -name *COPYRIGHT* -o -name CREDITS -o -name ChangeLog -o -name Changelog -o -name CHANGELOG -o -name CONTRIBUTORS -o -name *FAQ* -o -name FEATURES -o -name FILES -o -name HACKING -o -name History -o -name HISTORY -o -name INSTALL* -o -name LICENSE -o -name LSM -o -name MANIFEST -o -name NEWS -o -name *README* -o -name *Readme* -o -name SITES -o -name *RELEASE* -o -name RELNOTES -o -name THANKS -o -name TIPS -o -name TODO -o -name VERSION -o -name CONFIGURATION* -o -name GPLLicense`
+do
+	cp -a $i pkg/usr/share/doc/$pkgname-$pkgver/
+done
+
+# remove /usr/share/doc/$pkgname-$pkgver directory if empty
+if [ -z "`ls -1 pkg/usr/share/doc/$pkgname-$pkgver`" ]; then
+       rmdir pkg/usr/share/doc/$pkgname-$pkgver
+fi
+
+# remove /usr/share/doc/ directory if empty
+if [ -z "`ls -1 pkg/usr/share/doc`" ]; then
+       rmdir pkg/usr/share/doc
+fi
+
+# move /usr/share/info files to /usr/info
+if [ -d pkg/usr/share/info ]; then
+	mkdir -p pkg/usr/info
+	cp -a pkg/usr/share/info/* pkg/usr/info/
+	rm -rf pkg/usr/share/info
+fi
 
 # move /usr/share/man files to /usr/man
 if [ -d pkg/usr/share/man ]; then
